### PR TITLE
[srgssr] Extract the correct video + add subtitles (closes #14717)

### LINKED ISSUE #14717: [srgssr] Not all videos are extracted correctly + missing subtitle support
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.11.06*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.11.06**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Some [srgssr] videos are not extracted correctly, e.g. if we want to use youtube-dl on the video https://www.srf.ch/play/tv/tagesschau/video/150-jahre-reclam-buecher?id=5ec276f8-451b-4d60-9751-71e9e1edd7ef&station=69e8ac16-4327-4af4-b873-fd5cd6e895a7, it will extract the whole show https://www.srf.ch/play/tv/tagesschau/video/tagesschau-vom-10-11-2017-1930?id=96c125e5-e8c2-423f-a1a0-3638fa15d58e, and not the relevant segment of this show. 

To illustrate this, I append the youtube-dl logs of these two different videos:

```
[debug] System config: []
[debug] User config: ['--netrc']
[debug] Custom config: []
[debug] Command-line args: ['https://www.srf.ch/play/tv/tagesschau/video/tagesschau-vom-10-11-2017-1930?id=96c125e5-e8c2-423f-a1a0-3638fa15d58e', '-F', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.11.06
[debug] Python version 3.6.3 - Linux-4.13.11-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.4, ffprobe 3.4, rtmpdump 2.4
[debug] Proxy map: {}
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading JSON metadata
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading HTTP-HDS-HD token
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading f4m manifest
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading HTTP-HDS-SD token
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading f4m manifest
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading HTTP-HLS-SD token
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading m3u8 information
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading HTTP-HLS-HD token
[SRGSSR] 96c125e5-e8c2-423f-a1a0-3638fa15d58e: Downloading m3u8 information
[info] Available formats for 96c125e5-e8c2-423f-a1a0-3638fa15d58e:
format code       extension  resolution note
HTTP-HLS-HD-46    mp4        audio only   46k , mp4a.40.2
HTTP-HLS-SD-46    mp4        audio only   46k , mp4a.40.2
HTTP-HLS-HD-146   mp4        320x180     146k , avc1.66.30, mp4a.40.2
HTTP-HLS-SD-146   mp4        320x180     146k , avc1.66.30, mp4a.40.2
HTTP-HLS-HD-346   mp4        480x272     346k , avc1.66.30, mp4a.40.2
HTTP-HLS-SD-346   mp4        480x272     346k , avc1.66.30, mp4a.40.2
HTTP-HLS-HD-693   mp4        512x288     693k , avc1.77.30, mp4a.40.2
HTTP-HLS-SD-693   mp4        512x288     693k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-1325  mp4        640x360    1325k , avc1.77.30, mp4a.40.2
HTTP-HLS-SD-1325  mp4        640x360    1325k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-2125  mp4        960x544    2125k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-3626  mp4        1280x720   3626k , avc1.77.30, mp4a.40.2
HTTP-SD           mp4        unknown    
HTTP-HD           mp4        unknown    (best)
```

```
[debug] System config: []
[debug] User config: ['--netrc']
[debug] Custom config: []
[debug] Command-line args: ['https://www.srf.ch/play/tv/tagesschau/video/150-jahre-reclam-buecher?id=5ec276f8-451b-4d60-9751-71e9e1edd7ef', '-F', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.11.06
[debug] Python version 3.6.3 - Linux-4.13.11-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.4, ffprobe 3.4, rtmpdump 2.4
[debug] Proxy map: {}
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading JSON metadata
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading HTTP-HDS-HD token
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading f4m manifest
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading HTTP-HDS-SD token
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading f4m manifest
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading HTTP-HLS-SD token
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading m3u8 information
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading HTTP-HLS-HD token
[SRGSSR] 5ec276f8-451b-4d60-9751-71e9e1edd7ef: Downloading m3u8 information
[info] Available formats for 5ec276f8-451b-4d60-9751-71e9e1edd7ef:
format code       extension  resolution note
HTTP-HLS-HD-46    mp4        audio only   46k , mp4a.40.2
HTTP-HLS-SD-46    mp4        audio only   46k , mp4a.40.2
HTTP-HLS-HD-146   mp4        320x180     146k , avc1.66.30, mp4a.40.2
HTTP-HLS-SD-146   mp4        320x180     146k , avc1.66.30, mp4a.40.2
HTTP-HLS-HD-346   mp4        480x272     346k , avc1.66.30, mp4a.40.2
HTTP-HLS-SD-346   mp4        480x272     346k , avc1.66.30, mp4a.40.2
HTTP-HLS-HD-693   mp4        512x288     693k , avc1.77.30, mp4a.40.2
HTTP-HLS-SD-693   mp4        512x288     693k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-1325  mp4        640x360    1325k , avc1.77.30, mp4a.40.2
HTTP-HLS-SD-1325  mp4        640x360    1325k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-2125  mp4        960x544    2125k , avc1.77.30, mp4a.40.2
HTTP-HLS-HD-3626  mp4        1280x720   3626k , avc1.77.30, mp4a.40.2
HTTP-SD           mp4        unknown    
HTTP-HD           mp4        unknown    (best)
```

The logs are the same, this should not be the case!

Furthermore, it would be nice to have subtitle support, since almost all [srgssr] videos provide them.

I am working on solving these issues, you can expect a pull request soon.

### PR DESCRIPTION
**Pull request by goggle**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This pull request solves the issues described in #14717, and adds subtitle support. Furthermore, it uses `self._extract_akamai_formats` to extract these formats too, and the tests have been adjusted.

### CODE REVIEW

**Comment by refacto** [comment]
This would be awesome to have. Subtitles are currently broken for me with the srgssr extractor.

**Comment by goggle** [comment]
I think this needs some rework. We should not remove the old method to get the video sources... instead we should keep them as a fallback.

File: youtube_dl/extractor/srgssr.py [hunk 1]
```diff
@@ -33,102 +50,167 @@ def _get_tokenized_src(self, url, video_id, format_id):
...
+                    'ext': mimetype2ext(mime_type) if mime_type else None,
+                    'url': format_url,
+                    'preference': preference(quality)
+                })
+            podcast_keys = ('podcastSdUrl', 'podcastHdUrl')
+            podcast_qualities = ('SD', 'HD')
+
+        # This is needed because for audio medias the podcast url is usually
+        # always included, even if is only an audio segment and not the
+        # whole episode.
```
Comment (remitamine) [inline youtube_dl/extractor/srgssr.py:156]: do you have an example for this case(whole podcast URL present for an audio segment)?

File: youtube_dl/extractor/srgssr.py [hunk 2]
```diff
@@ -33,102 +50,167 @@ def _get_tokenized_src(self, url, video_id, format_id):
...
+
+        subtitle_data = media_data.get('subtitleList', [])
+        default_language_codes = {
+            'srf': 'de',
+            'rts': 'fr',
+            'rsi': 'it',
+            'rtr': 'rm',
+            'swi': 'en',
+        }
+        known_formats = ('TTML', 'VTT')
```
Comment (remitamine) [inline youtube_dl/extractor/srgssr.py:91]: do you have an example with TTML subtitles? all the videos that i've tested with has only VTT subtitles.

Comment (goggle) [reply youtube_dl/extractor/srgssr.py:91]: I think TTML subtitles do not exist anymore.

This video (https://www.srf.ch/play/tv/rundschau/video/schwander-rot-gruene-stadtpolitik-min-li-marti-tamilen-kirche?urn=urn:srf:video:2da578e3-dbb4-4657-a539-f01089a67831) used to have TTML subtitles.

**Pull request closed without merging**